src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,28 @@
+package duke.task;
+
+import duke.Storage;
+
+import java.time.LocalDateTime;
+
+/**
+ * Deadline represents a task with an end date.
+ */
+public class Deadline extends Task {
+    private LocalDateTime deadline;
+
+    public Deadline(String taskName, String deadline) {
+        super(taskName);
+        this.deadline = Storage.stringToDate(deadline);
+    }
+
+    @Override
+    public String toDatabaseRepresentation() {
+        return String.format("D | %s | %s | %s",
+            isDone() ? 1 : 0, getTaskName(), Storage.dateToDatabaseRepresentation(this.deadline));
+    }
+
+    @Override
+    public String toString() {
+        return String.format("[D]%s (by: %s)", super.toString(), Storage.dateToString(this.deadline));
+    }
+}
```

Comment:
I like that your string representation methods for Todo, Deadline and Event use String.format(). This looks a lot more elegant than hardcoding and something that I can try to do in my iP.